backend/ccm/canvas_api/admin_sections_api_handler.py

```diff
@@ -0,0 +1,189 @@
+            courseQueryParams['search_term'] = course_name
+            logger.info(f"Searching for courses with name: {course_name}")
+        
+        # Get sections from all courses for filtered accounts
+        courses_with_sections_results = []
+        courses_with_sections_errors = []
+        time_fetch_all_sections_start: float = time.perf_counter()
+        async with asyncio.TaskGroup() as tg:
+            for account_id in filtered_account_ids:
+                account = accounts_by_account_id[account_id]
+                tg.create_task(
+                    fetch_sections_for_account_task(
+                        account,
+                        courseQueryParams,
+                        self.courses_allowed_fields,
+                        self.sections_allowed_fields,
+                        courses_with_sections_results, 
+                        courses_with_sections_errors)
+                )
+        time_fetch_all_sections_end: float = time.perf_counter()
+        logger.info(f"TIME taken to fetch sections for all accounts: {time_fetch_all_sections_end - time_fetch_all_sections_start:.2f} seconds")
+
+        if courses_with_sections_errors:
+            self.canvas_error.handle_canvas_api_exceptions(courses_with_sections_errors)
+            error_response = self.canvas_error.to_dict()
+            return Response(error_response, status=error_response.get('statusCode'))
+        
+        end_time: float = time.perf_counter()
+        logger.info(f"TIME TOTAL taken to get sections: {end_time - start_time:.2f} seconds")
+        return Response(courses_with_sections_results, status=HTTPStatus.OK)
+    
+async def retrieve_accounts_map(canvas_api: Canvas) -> dict:
+    """
+    Retrieve accounts accessible to the admin user.
+    """
+    accounts_by_account_id = {}
+    time_get_accounts_start: float = time.perf_counter()
+    accounts = await asyncio.to_thread(canvas_api.get_accounts)
+    time_get_accounts_end: float = time.perf_counter()
+    logger.info(f"TIME taken to get accounts: {time_get_accounts_end - time_get_accounts_start:.2f} seconds")
+    for account in accounts:
+        logger.info(f"Account ID: {account.id}, Name: {account.name}, Parent Account ID: {account.parent_account_id}")
+        accounts_by_account_id[account.id] = account
+    return accounts_by_account_id
+
+async def fetch_sections_for_account_task(
+        account: Account, 
+        courseQueryParams: dict, 
+        courses_allowed_fields: set,
+        sections_allowed_fields: set,
+        results: list,
+        errors: list
+    ) -> list:
+    # First retrieve cour+ses for the account
+    courses = []
+    time_get_courses_start: float = time.perf_counter()
+    try:
+        def get_courses(courseQueryParams):
+            return account.get_courses(**courseQueryParams)
+        logger.info(f"Retrieving courses for account_id: {account.id}")
+        courses = await asyncio.to_thread(get_courses, courseQueryParams)
```

Comment:
What is the need for the nested functional call?

 will it work if you extract the below to a separate function?  felt that could simply thing for readability, again i understanding the purpose without my opinion 
 
 try:
        def get_courses(courseQueryParams):
            return account.get_courses(**courseQueryParams)
    except: 
        pass